inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java

```diff
@@ -136,14 +168,33 @@ private ClsClient getClsClient(ClsDataNodeDTO clsDataNode) {
     /**
      * Create topic index by tokenizer
      */
-    private void createTopicIndex(SinkInfo sinkInfo) throws BusinessException {
+    private void createTopicIndex(SinkInfo sinkInfo, ClsDataNodeDTO clsDataNode) throws BusinessException {
+
         ClsSinkDTO clsSinkDTO = JsonUtils.parseObject(sinkInfo.getExtParams(), ClsSinkDTO.class);
-        if (StringUtils.isNotBlank(clsSinkDTO.getTokenizer())) {
+        if (StringUtils.isEmpty(clsSinkDTO.getTokenizer())) {
             LOG.warn("topic {} tokenizer is empty", clsSinkDTO.getTopicName());
             return;
         }
-        ClsDataNodeDTO clsDataNode = getClsDataNode(sinkInfo);
+        FullTextInfo topicIndexFullText = getTopicIndexFullText(sinkInfo, clsDataNode);
+        if (ObjectUtils.anyNotNull(topicIndexFullText)) {
+            // if topic index exist,update
+            updateTopicIndex(sinkInfo, clsDataNode);
+            return;
+        }
         ClsClient clsClient = getClsClient(clsDataNode);
+        CreateIndexRequest req = getCreateIndexRequest(clsSinkDTO);
+        try {
+            clsClient.CreateIndex(req);
+        } catch (TencentCloudSDKException e) {
+            String errMsg = "Create cls topic index failed: " + e.getMessage();
+            LOG.error(errMsg, e);
+            sinkService.updateStatus(sinkInfo.getId(), SinkStatus.CONFIG_FAILED.getCode(), errMsg);
+            throw new BusinessException(errMsg);
+        }
+        LOG.info("topic {} create index success tokenizer is {}", clsSinkDTO.getTopicName(), clsSinkDTO.getTokenizer());
```

Comment:
create index success for topic ={}, tokenizer = {}

Reply:
done